boards/arm/frdm_k64f/pinmux.c

```diff
@@ -33,21 +33,21 @@ static int frdm_k64f_pinmux_init(struct device *dev)
 		device_get_binding(CONFIG_PINMUX_MCUX_PORTE_NAME);
 #endif
 
-#ifdef CONFIG_UART_MCUX_0
+#ifdef DT_NXP_KINETIS_UART_4006A000
```

Comment:
Will using address instead of index make the drivers more confusing to read when using the datasheet as the reference material? I personally like this approach (using address) but I am asking in general

Reply:
On my side, I'd really prefer using label instead of address. Reference materiel most generally refers to the human friendly names of the instances (equivalent of dts "label").
The need for knowing the address makes code reading/review much more difficult.
dts provides ways to associate and address with an address, why can't we rely on that to provide human/doc friendly definitions?

Besides, if this is required in non board specific code (seldom, but could happen), it will make quite complex ifdefery (since we'll have to consider all possible addresses).

Reply:
I get your point, `instance-name`, `device-name` would also do if there is no equivalent in Linux yet. 
So, if I'm correct only places where `label` use is correct is currently for buttons and leds. And all other `label` (roughly under `dts/`) could be replaced by a better alternative, right?

Reply:
@galak I would appreciate if you could clarify: how this would play nicely, if we have SOCs with Secure and Non-Secure domains, which will normally have different address labels. Or do I miss something?

Reply:
@erwango yes, exactly.  I don't think Linux has anything equivalent since in normal use the device tree path would to identify the node.  `zephyr,name` would be a good choice as it will clearly not conflict with any existing property use and the `zephyr,` prefix could be stripped by the tooling before generating the macros.

Reply:
I'd also prefer to use labels (but the node labels, not the "label" properties) instead of addresses for identifying instances. We use the labels for referring the specific instance nodes in board dts files and overlays, so I think such approach would be consistent.  Though, I'm not sure how the generated symbols should look like. Something like `DT_NXP_KINETIS_UART_UART0` is not very pretty, especially that for single instances we'll end up with for example `DT_NXP_KINETIS_WDG_WDG`. Maybe `DT_NXP_KINETIS_UART_INSTANCE_UART0` then?
And when it comes to instance addresses, it seems there is no need to generate a dedicated symbol for this purpose. Why `#ifdef DT_NXP_KINETIS_UART_4006A000_BASE_ADDRESS` would not be sufficient in this case?

Reply:
I think we should be careful with 'instance/INSTANCE' usage following recent #define introductions.
Now we have:
 - compute time instances. Starting from _0, defined at compilation depending on what is enabled on target and sorted in a arbitrary order
 - datasheet instances. Might not start from _0,  static and currently reflected in dts `label` property.
Maybe we can use `device` for the later from now on.